src/compiler/chainrules.jl

```diff
@@ -115,28 +115,60 @@ for T_outer in (:Tuple, :NamedTuple)
     ChainRulesCore.backing(xp)  # this is accessing ChainRulesCore internals, but it is prob safe enough, and it is fastest
   end
 end
-# Could `reinterpret` instead of broadcasting here -- TODO
-@inline wrap_chainrules_output(xs::AbstractArray{<:ChainRules.Tangent}) = wrap_chainrules_output.(xs)
+wrap_chainrules_output(dxs::AbstractArray{<:Number}) = dxs
+wrap_chainrules_output(dxs::AbstractArray{<:AbstractArray{<:Number}}) = dxs
```

Comment:
Is there an mwe for this being needed? Is it for RecursiveArrayTools or does it come up elsewhere?

Reply:
For all of these, the minimal correct thing is `wrap_chainrules_output(dxs::AbstractArray) = map(wrap_chainrules_output, dxs)`. 

But for some common types, we know that the representation used by Zygote and by ChainRules agrees, so we can save a copy by doing nothing. That's the case for `AbstractArray{<:Number}` and `AbstractArray{<:AbstractArray{<:Number}}`.